### ISSUE TITLE
After applying #23480 patch to 9.0, usernames disappear from Calendar sharing

### ISSUE THREAD

**Issue by devuan2** [owncloud/core#23864]
### Steps to reproduce
1. Apply patch #23480 
2. Login as LDAP user
3. Share calendar
4. "hashed/UUID" usernames appear instead of human readable username

in 8.x, this used to use the "displayName" attribute from LDAP iirc..
Screenshot here: http://imgur.com/r8G1Szp
### Expected behaviour

Actual user name should display, not UUID
### Actual behaviour

UUID displays
### Server configuration

**Operating system**: 
Linux Ubuntu 14.04 x86_64

**Web server:** 
Apache 2.4.7-1ubuntu4.9

**Database:** 
mysql 5.5.47-0ubuntu0.14.04.1

**PHP version:**
5.5.9+dfsg-1ubuntu4.14

**ownCloud version:** 
9.0

**Updated from an older ownCloud or fresh install:**
Updated

**Where did you install ownCloud from:**
http://download.owncloud.org/download/repositories/stable/Ubuntu_14.04/

**Signing status (ownCloud 9.0 and above):**
wut?

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

No errors have been found.

**List of activated apps:**

```
 Activity 2.2.1
by Frank Karlitschek, Joas Schilling (AGPL-licensed)
Official
Show description …

Announcement Center 1.1.1
by Joas Schilling (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Collaborative tags 0.2
by Vincent Petry (AGPL-licensed)
Official
Show description …

Comments 0.2
by Arthur Shiwon, Vincent Petry (AGPL-licensed)
Official
Show description …

Deleted files 0.8.0
by Bjoern Schiessle (AGPL-licensed)
Official
Show description …

Federation 0.0.4
by Bjoern Schiessle (AGPL-licensed)
Official
Show description …

First Run Wizard 1.1
by Frank Karlitschek, Jan-Christoph Borchardt (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Gallery 14.5.0
by Olivier Paroz, Robin Appelman (AGPL-licensed)
Official
Show description …
Enable only for specific groups
LDAP user and group backend 0.8.0
by Dominik Schmidt and Arthur Schiwon (AGPL-licensed)
Official
Show description …

Mail Template Editor 0.1
by Jörn Dreyer (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Notifications 0.2.3
by Joas Schilling (AGPL-licensed)
Official
Show description …

PDF Viewer 0.8
by Thomas Müller, Lukas Reschke (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Provisioning API 0.4.1
by Tom Needham (AGPL-licensed)
Official
Show description …

Share Files 0.9.1
by Michael Gapczynski, Bjoern Schiessle (AGPL-licensed)
Official
Show description …

Text Editor 2.1
by Tom Needham, Björn Schießle (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Update notification 0.1.0
by Lukas Reschke (AGPL-licensed)
Official
Show description …
Enable only for specific groups
Versions 1.2.0
by Frank Karlitschek, Bjoern Schiessle (AGPL-licensed)
Official
Show description …

Video player 0.9.8
by Björn Korella (MIT-licensed)
Official
Show description …
Enable only for specific groups
Calendar 1.0
by Georg Ehrke, Raghu Nayyar, Bernhard Fröhler (AGPL-licensed)
Approved
Show description …
Enable only for specific groups
Contacts 1.0.0.0
by Hendrik Leppelsack, Thomas Müller, Alexander Weidinger, Jan-Christoph Borchardt, Tom Needham (AGPL-licensed)
Approved
Show description …

This app has an update available.
Enable only for specific groups
Documents 0.12.0
by Frank Karlitschek, Victor Dubiniuk (AGPL-licensed)
Approved
Show description …
Enable only for specific groups
Tasks 0.9.0
by Raimund Schlüßler (AGPL-licensed)
Approved
Show description …
Enable only for specific groups
Experimental applications ahead

Experimental apps are not checked for security issues, new or known to be unstable and under heavy development. Installing them can cause data loss or security breaches.
OwnNote 1.07
by Ben Curtis (AGPL-licensed)
Experimental
```

**The content of config/config.php:**

``` php
<?php
$CONFIG = array (
  'instanceid' => 'ocwe0wi2aumw',
  'passwordsalt' => 'removed',
  'secret' => 'removed',
  'trusted_domains' => 
  array (
    0 => 'example.com',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'https://example.com',
  'dbtype' => 'mysql',
  'version' => '9.0.0.19',
  'dbname' => 'bobbytables',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'removed',
  'dbpassword' => 'removed',
  'logtimezone' => 'UTC',
  'installed' => true,
  'mail_from_address' => 'root',
  'mail_smtpmode' => 'sendmail',
  'mail_domain' => 'example.com',
  'ldapIgnoreNamingRules' => false,
  'updatechecker' => false,
  'theme' => '',
  'loglevel' => 2,
  'maintenance' => false,
  'debug' => false,
);
```

``` json
{
    "system": {
        "instanceid": "ocwe0wi2aumw",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "example.com"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/example.com",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "bobbytables",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "root",
        "mail_smtpmode": "sendmail",
        "mail_domain": "removed",
        "ldapIgnoreNamingRules": false,
        "updatechecker": false,
        "theme": "",
        "loglevel": 2,
        "maintenance": false,
        "debug": false
    }
}


**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
OpenLDAP

#### LDAP configuration (delete this part if not used)

```

+-------------------------------+----------------------------------------------------------------------------------------------------+
| Configuration                 | s01                                                                                                |
+-------------------------------+----------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      |                                                                                                    |
| hasPagedResultSupport         |                                                                                                    |
| homeFolderNamingRule          |                                                                                                    |
| lastJpegPhotoLookup           | 0                                                                                                  |
| ldapAgentName                 |                                                                                                    |
| ldapAgentPassword             | **\*                                                                                                |
| ldapAttributesForGroupSearch  |                                                                                                    |
| ldapAttributesForUserSearch   |                                                                                                    |
| ldapBackupHost                |                                                                                                    |
| ldapBackupPort                |                                                                                                    |
| ldapBase                      | dc=example,dc=com                                                                           |
| ldapBaseGroups                | dc=example,dc=com                                                                           |
| ldapBaseUsers                 | dc=example,dc=com                                                                           |
| ldapCacheTTL                  | 600                                                                                                |
| ldapConfigurationActive       | 1                                                                                                  |
| ldapDynamicGroupMemberURL     |                                                                                                    |
| ldapEmailAttribute            | mail                                                                                               |
| ldapExperiencedAdmin          | 0                                                                                                  |
| ldapExpertUUIDGroupAttr       |                                                                                                    |
| ldapExpertUUIDUserAttr        |                                                                                                    |
| ldapExpertUsernameAttr        |                                                                                                    |
| ldapGroupDisplayName          | cn                                                                                                 |
| ldapGroupFilter               | (|(cn=owncloud))                                                                                   |
| ldapGroupFilterGroups         | owncloud                                                                                           |
| ldapGroupFilterMode           | 0                                                                                                  |
| ldapGroupFilterObjectclass    |                                                                                                    |
| ldapGroupMemberAssocAttr      | member                                                                                             |
| ldapHost                      | ldaps://ldap1.example.com                                                                      |
| ldapIgnoreNamingRules         |                                                                                                    |
| ldapLoginFilter               | (&(objectClass=posixAccount) (memberOf=CN=owncloud,OU=groups,dc=example,dc=com) (uid=%uid)) |
| ldapLoginFilterAttributes     |                                                                                                    |
| ldapLoginFilterEmail          | 0                                                                                                  |
| ldapLoginFilterMode           | 1                                                                                                  |
| ldapLoginFilterUsername       | 1                                                                                                  |
| ldapNestedGroups              | 0                                                                                                  |
| ldapOverrideMainServer        |                                                                                                    |
| ldapPagingSize                | 500                                                                                                |
| ldapPort                      | 636                                                                                                |
| ldapQuotaAttribute            |                                                                                                    |
| ldapQuotaDefault              |                                                                                                    |
| ldapTLS                       | 0                                                                                                  |
| ldapUserDisplayName           | displayName                                                                                        |
| ldapUserDisplayName2          |                                                                                                    |
| ldapUserFilter                | (memberOf=CN=owncloud,OU=groups,dc=example,dc=com)                                          |
| ldapUserFilterGroups          |                                                                                                    |
| ldapUserFilterMode            | 1                                                                                                  |
| ldapUserFilterObjectclass     | inetOrgPerson                                                                                      |
| ldapUuidGroupAttribute        | auto                                                                                               |
| ldapUuidUserAttribute         | auto                                                                                               |
| turnOffCertCheck              | 0                                                                                                  |
| useMemberOfToDetectMembership | 1                                                                                                  |
+-------------------------------+----------------------------------------------------------------------------------------------------+

```


### Client configuration
**Browser:**
firefox 45.0.1

**Operating system:**
Devuan Linux x86_64

### Logs
#### Web server error log
```

Insert your webserver log here

```

#### ownCloud log (data/owncloud.log)
```

{"reqId":"Uo29RhLIyhJtGWjd15e8","remoteAddr":"xxx.xxx.xxx.176","app":"OCP\Share","message":"Sharing backend for principals not found","level":3,"time":"2016-04-08T13:37:20+00:00"}

```


#### Browser log
```

n/a

```
```

**Comment by DeepDiver1975** [comment]
@blizzz mind testing this? THX

**Comment by blizzz** [comment]
The actual username is what is on the very left column on the users page.

**Comment by nickvergessen** [comment]
Was this tested with 9.0.0 or 9.0.1 @devuan2

**Comment by devuan2** [comment]
@nickvergessen  Appears to be Version: 9.0.0-2.1 (installed via apt from 
http://download.owncloud.org/download/repositories/stable/Ubuntu_14.04/).   `aptitude search '~U'` is showing three owncloud updates waiting to be applied so I will do that in the next couple days.

**Comment by devuan2** [comment]
Hrm.. upgrade didn't go so well. http://imgur.com/sqW2YT1... trying to post owncloud.log but getting "Something went really wrong, and we can’t process that file. .."

**Comment by devuan2** [comment]
..logfile..[upgrade.txt](https://github.com/owncloud/core/files/217834/upgrade.txt)

**Comment by devuan2** [comment]
Was able to apply the 9.0.1-1.1 update today. Some of the apps ended up uninstalled (Calendar, Tasks) but after reinstalling them, I seem to have everything running on the latest 9.0 update. Still have the original problem of calendar sharing showing the hashed username though.

**Comment by devuan2** [comment]
Looks like this issue is limited to the calendar app only. Sharing Docs or Pics displays the correct (Full Name) column from the users page. I've updated to version 1.2.2 of the calendar app and this is still broken.

I've tried poking around in owncloud/apps/calendar/templates/part.calendarlist.item.php but don't really know what I'm looking for to display the correct column. line 109 references `userShare.displayname` but no idea if that's the right place to be hacking around to fix this. Any suggestions?
